core/src/test/scala/org/apache/spark/scheduler/FakeTask.scala

```diff
@@ -36,34 +36,71 @@ class FakeTask(
   override def preferredLocations: Seq[TaskLocation] = prefLocs
 }
 
+class FakeLongTasks(
+    stageId: Int,
+    partitionId: Int,
+    prefLocs: Seq[TaskLocation] = Nil,
+    serializedTaskMetrics: Array[Byte] =
+    SparkEnv.get.closureSerializer.newInstance().serialize(TaskMetrics.registered).array(),
+    isBarrier: Boolean = false)
+  extends FakeTask(stageId, partitionId, prefLocs, serializedTaskMetrics, isBarrier) {
+
+  override def runTask(context: TaskContext): Int = {
+    while (true) {
+      Thread.sleep(10000)
+    }
+    0
+  }
+}
+
 object FakeTask {
   /**
    * Utility method to create a TaskSet, potentially setting a particular sequence of preferred
    * locations for each task (given as varargs) if this sequence is not empty.
    */
   def createTaskSet(numTasks: Int, prefLocs: Seq[TaskLocation]*): TaskSet = {
-    createTaskSet(numTasks, stageId = 0, stageAttemptId = 0, priority = 0, prefLocs: _*)
+    createTaskSet(numTasks, 0, 0, 0, false, prefLocs: _*)
```

Comment:
Can we preserve the existing `stageId = 0` style? It seems that you remove them to reduce the line length.

Reply:
Sure. Seems right, i have updated as per the same